YunoHost/yunohost
# WIP: rewrite the CLI/API interface using Typer+FastAPI

## POC of using typer + fastapi for managing cli commands + api endpoints


## PR Status
Very POC
- simple commands ok
- type checking ok for basic types
- help from docstring

## Todo
- custom prompt

## How to test

```
# To test api
systemctl stop yunohost-api
yunohost-fastapi --reload --reload-dirs /ynh-dev/yunohost
```
Go to https://{LXC_IP}/yunohost/api/docs

```
for cli
yunohost-typer --help
```

regular `yunohost` and `yunohost-api` should still ~work to compare with
